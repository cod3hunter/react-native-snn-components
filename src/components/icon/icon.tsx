import React from 'react';
import { StyleSheet, Text } from 'react-native';
import Styles from './icon.styles';
import { useTheme } from '../../hooks/theme/useTheme';

export type IconProps = {
  name: string;
  size: number;
  style?: StyleSheet.NamedStyles<unknown>;
  color: string;
};

let FeatherIcon: any;

try {
  FeatherIcon = require('react-native-vector-icons/Feather').default;
} catch (e) {
  let isErrorLogged = false;

  FeatherIcon = ({ name, color, size, ...rest }: IconProps) => {
    const [styles] = useTheme(Styles);
    if (!isErrorLogged) {
      if (
        !/(Cannot find module|Module not found|Cannot resolve module|Requiring unknown module)/.test(
          (e as any).message
        )
      ) {
        console.error(e);
      }

      console.warn(
        `Tried to use the icon '${name}' in a component from 'react-native-snn-components', but 'react-native-vector-icons/Feather' could not be loaded.`,
        `To remove this warning, try installing 'react-native-vector-icons'.`
      );

      isErrorLogged = true;
    }

    return (
      <Text
        {...rest}
        style={[styles.icon, { color, fontSize: size }]}
        // @ts-expect-error
        pointerEvents="none"
        selectable={false}
      >
        â–¡
      </Text>
    );
  };
}

export const Icon = ({ name, size, color, style }: IconProps) => {
  return <FeatherIcon {...{ name, size, color, style }} />;
};
