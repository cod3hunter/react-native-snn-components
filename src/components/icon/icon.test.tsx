console.error = () => {};
console.warn = () => {};

import React from 'react';
import { render } from '@testing-library/react-native';
import { Icon } from './icon';
it('render icon', async () => {
  const { toJSON } = render(<Icon name="plus" color="green" size={12} />);
  expect(toJSON()).toMatchSnapshot();
});
