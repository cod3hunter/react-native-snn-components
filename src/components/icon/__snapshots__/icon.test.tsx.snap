// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render icon 1`] = `
<Text
  pointerEvents="none"
  selectable={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "color": "green",
        "fontSize": 12,
      },
    ]
  }
  testID="feather-icon"
>
  â–¡
</Text>
`;
